project1/src/main/java/uk/co/cpsd/javaproject1/WorldPanel.java

```diff
@@ -36,25 +36,23 @@ protected void paintComponent(Graphics g) {
             }
         }
         
+        Font font=new Font("Arial", Font.BOLD, Math.max(10, cellSize / 3));
```

Comment:
You're still creating a new font every time you paint.  Can you not create the font once in the constructor?